packages/active-record/src/-private/builders/query.ts

```diff
@@ -76,12 +76,14 @@ export function query(
 
   copyForwardUrlOptions(urlOptions, options);
 
-  const url = buildBaseURL(urlOptions);
+  const url = new URL(buildBaseURL(urlOptions));
+  url.search = buildQueryParams(query, options.urlParamsSettings);
+
   const headers = new Headers();
   headers.append('Accept', 'application/json;charset=utf-8');
 
   return {
-    url: `${url}?${buildQueryParams(query, options.urlParamsSettings)}`,
```

Comment:
I think instead we should check if there is a query / check if buildQueryParams has output and use that to conditionally join with a `?`